inky/inky.go

```diff
@@ -148,13 +194,35 @@ func New(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, o *Opts
 	case PHAT:
 		d.bounds = image.Rect(0, 0, 104, 212)
 		d.flipVertically = true
+	case PHAT2:
+		d.bounds = image.Rect(0, 0, 122, 250)
+		d.flipVertically = true
 	case WHAT:
 		d.bounds = image.Rect(0, 0, 400, 300)
 	}
 
 	return d, nil
 }
 
+// NewDetected tries to open a handle to an Inky pHat or wHAT automatically detected from EEPROM
+func NewDetected() (*Dev, error) {
+	opts, err := DetectOpts("1")
+	if err != nil {
+		return nil, err
+	}
+
+	spiPort, err := spireg.Open("SPI0.0")
```

Comment:
Do not hardcode spi ports, this can change wildly, e.g. someone connecting it over USB on Windows.